airflow-core/src/airflow/dag_processing/dagbag.py

```diff
@@ -706,4 +716,5 @@ def sync_bag_to_db(
         None,  # file parsing duration is not well defined when parsing multiple files / multiple DAGs.
         dagbag.dag_warnings,
         session=session,
+        files_parsed=files_parsed if files_parsed else None,
```

Comment:
Can we not just do `files_parsed=files_parsed`? Or is None special to an empty set?